### PR TITLE
Make it easy to leverage rhoas code in other CLIs

### PR DESCRIPTION
**Pull request by lburgazzoli**
- feat: make it possible to configure the config/context file location
- chore: remove useless checks

Closes # 

### Verification Steps

1. Do x
2. Do y

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@wtrocki I've opened this PR mainly for discussion and exploration as the goal here is to make the code more reusable by other CLIs. Let me know what you think.

**Comment by wtrocki** [comment]
This is very good! I really like how this is shaping towards more flexible version of the SDK.
We have more work planned on plugin side post sprint but this PR provides good base for us!

File: pkg/core/configutil/file.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+	path = filepath.Join(dir, defaultFileName)
+	if err != nil {
+		return "", err
+	}
+
+	return path, nil
+}
+
+func (c *ConfigFile) defaultDir() (string, error) {
```
Comment (rkpattnaik780) [inline pkg/core/configutil/file.go:168]: Context and config file seem to be using the same file now, authentication and context operations seem to be undoing each other.

**Comment by rkpattnaik780** [comment]
This will really simplify stuff for plugin architecture!
Thanks!

File: pkg/core/configutil/file.go (hunk 1 above)
Comment (lburgazzoli) [reply pkg/core/configutil/file.go:168]: will fix it

Comment (lburgazzoli) [reply pkg/core/configutil/file.go:166]: fixed

**Comment by lburgazzoli** [comment]
@wtrocki @rkpattnaik780 is anything you want me to address ?

**Comment by rkpattnaik780** [comment]
Hi @lburgazzoli !

Noticed a slight anomally with status command. Seems the contexts arent loaded from default file. (It is unchanged in the status command too).

```
> ./rhoas context status              
Service Context Name:   rama-test
Context File Location:  /home/rkpattnaik780/.config/rhoas/context.json

  Service Registry
  ------------------------------------------------------------------------------------------------------
  ID:                    31f0cd22-7b7c-46ac-84aa-3342bbdeb351
  Name:                  rama-test
  Status:                ready
  Registry URL:          https://bu98.serviceregistry.rhcloud.com/t/31f0cd22-7b7c-46ac-84aa-3342bbdeb351

➜  cli git:(PR_1556) cat /home/rkpattnaik780/.config/rhoas/contexts.json
{
  "contexts": {
    "dev": {
      "kafkaID": "c9sa7pil3jfoaj678280",
      "serviceregistryID": "cc8a243a-feed-4a4c-9394-5a35ce83cca5"
    },
    "my-context": {
      "kafkaID": "c9sa7pil3jfoaj678280",
      "serviceregistryID": ""
    }
  },
  "current_context": "my-context"
}
```

**Comment by lburgazzoli** [comment]
> Noticed a slight anomally with status command. Seems the contexts arent loaded from default file. (It is unchanged in the status command too).
> 

can you tell me the steps you followed so I can reproduce it ?

**Comment by rkpattnaik780** [comment]
> can you tell me the steps you followed so I can reproduce it ?

1. Run `rhoas context list`
2. Get location of context file using the status command.
```
rhoas context status
```
3. View the context file

**Comment by wtrocki** [comment]
We have moved with the implementation and adopted some of the changes (not all) There is a lot of conflicts in this PR so we can close it and look for future feedback on what is not working in rhoas cli sdk

**Pull request closed without merging**